### ISSUE TITLE
Improve rendering of aeroways

### ISSUE THREAD

**Issue by endim8** [organicmaps/organicmaps#2078]
https://wiki.openstreetmap.org/wiki/Aeroways

I think the main improvements would be filling `aeroway=runway`,  thickening `aeroway=taxiway` and filling `aeroway=helipad` as these are the most prominent and most used elements.

<img width="1479" alt="aeroway" src="https://user-images.githubusercontent.com/26939824/153515012-ff0ecfc4-9377-40f3-9f48-25d45f670d0a.png">

**Comment by endim8** [comment]
I'm working on adding this, but I'm running into this issue when there are curved lines with large width values in the mapcss, is there any workaround for this?
<img width="215" alt="Screenshot 2022-02-17 at 12 11 31" src="https://user-images.githubusercontent.com/26939824/154479305-2529aca4-dbdf-49bc-93c2-c9c59812edff.png">

**Comment by biodranik** [comment]
We use outdated Kothic implementation which does not support all mapcss features properly. Also, our styles engine doesn't support all features (yet).

**Comment by pastk** [comment]
> I'm also encountering another issue, according to the [MapCSS 0.2 specification](https://wiki.openstreetmap.org/wiki/MapCSS/0.2), the `line` type should filter out ways with the same start/end node and ways with `area=yes`. This doesn't seem to be working, as the runway area is also inheriting the line style.
>
> This issue also seems to affect normal roads with `area=yes`: (9C3XCCGP+WW)
> 

@vng I guess line / area types distinction is done in the generator?

**Comment by vng** [comment]
Please, clarify what are you trying to fix in styles and what goes wrong with concrete type?

If osm element is closed (start==end) and we have _area (or node)_ rules for it, the object will be treated as area and all line 
rules will be skipped.

If osm element is closed (start==end) and we have _only line_ rules for it, the object will be treated as line.

**Comment by endim8** [comment]
> Please, clarify what are you trying to fix in styles and what goes wrong with concrete type?
> 
> If osm element is closed (start==end) and we have _any area_ rule for it, the object will be treated as area and all line 
> rules will be skipped.
> 
> If osm element is closed (start==end) and we have _only line_ rules for it, the object will be treated as line.

Ah i see! Thank you for the clarification, ill try it when i get back :)

**Comment by endim8** [comment]
Should area styles for roads be in `roads.mapcss` or `basemap.mapcss`?

**Comment by endim8** [comment]
Is `aeroway=runway` an area in the generator? I think that might be the issue

**Comment by pastk** [comment]
Yes it seems maps should be regenerated, see https://github.com/organicmaps/organicmaps/pull/2013#issuecomment-1050028898

**Comment by endim8** [comment]
> Yes it seems maps should be regenerated, see [#2013 (comment)](https://github.com/organicmaps/organicmaps/pull/2013#issuecomment-1050028898)

ooh, so once it's in the stylesheet it's automatically added to the generator and included in the MWM?

**Comment by pastk** [comment]
Yes the features that are not used in the stylesheets are omitted from the mwm files - this is my understanding.

@vng has the best knowledge of this.

**Comment by endim8** [comment]
right i see 👍 thank you :)

**Comment by endim8** [comment]
@pastk how do you usually work on styles for types that aren't yet included in the mwm? i'm finding it a bit difficult

**Comment by pastk** [comment]
> @pastk how do you usually work on styles for types that aren't yet included in the mwm? It seems like a bit of a chicken and egg situation

I had this situation only once so far - in https://github.com/organicmaps/organicmaps/pull/2013#issuecomment-1029491553
The change was quite obvious and not very big so it was accepted as is.
But ideally such changes should be tested by generating some small region, see [the docs](https://github.com/organicmaps/organicmaps/tree/master/tools/python/maps_generator/README.md) - I never did it so I can't advise if its difficult or not.

**Comment by pastk** [comment]
Which area-related change do you want to make exactly?
[It seems](https://wiki.openstreetmap.org/wiki/Tag:aeroway=runway) that `aeroway=runway` is seldom used as an area (4.6% cases) and the use of `area:aeroway=runway` is growing for this purpose (OM doesn't support it at all at the moment), but it wasn't formally decided (see also [discussion](https://wiki.openstreetmap.org/wiki/Talk:Tag:aeroway%3Drunway#Runways_as_areas))

**Comment by pastk** [comment]
- related #3214
